File: .framework/node/backend/lib/event.js
const axiosLib = require("axios");
const fs = require("fs");

const WILCO_ID = process.env.WILCO_ID || fs.readFileSync('../.wilco', 'utf8')

const baseURL = 'http://localhost:3003/';
const axios = axiosLib.create({
  baseURL: baseURL,
  headers: {
    'Content-type': 'application/json',
  },
});

async function sendEvent(event, metadata) {
  try {
    const result = await axios.get(`/hello`, JSON.stringify({event, metadata}));
    // const result = await axios.post(`/users/${WILCO_ID}/event`, JSON.stringify({event, metadata}));
    return result?.data;
  } catch (e) {
    // pass
  }
}

module.exports = {
  sendEvent,
}

File: .framework/rails/backend/lib/event.rb
module Event
  def sendEvent(eventName, metadata)
    wilcoId = ENV['WILCO_ID'] || File.read(Rails.root.join("../.wilco"))
    baseUrl =  ENV['ENGINE_BASE_URL'] || "https://engine.wilco.gg"
    conn = Faraday.new(
      url: baseUrl + "/users/#{wilcoId}/",
      headers: {'Content-Type' => 'application/json'}
    )

    response = conn.post('event') do |req|
      req.headers['Content-Type'] = 'application/json'
      req.body = { event: eventName, metadata: metadata}.to_json
    end
    response
  end
end
